pkg/provider/pods.go

```diff
@@ -202,3 +202,7 @@ func (p *Pod) CreatedByDeploymentConfig() (bool, error) {
 	}
 	return false, nil
 }
+
+func (p *Pod) HasNodeSelector() bool {
+	return p.Spec.NodeSelector != nil
```

Comment:
Should we consider `nodeName` as well?
 https://kubernetes.io/docs/concepts/scheduling-eviction/assign-pod-node/

Reply:
Good call I'll add this to the func as well.